core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -533,4 +531,51 @@ private Pair<InclusiveMetricsEvaluator, StrictMetricsEvaluator> metricsEvaluator
       return metricsEvaluators.get(partition);
     }
   }
+
+  private class FilesToDeleteHolder {
```

Comment:
Is there any way we can do this differently? In theory, we can add another abstract method, similar to how we handle manifest writers.

```
protected abstract Set<F> newFileSet();

protected abstract ManifestWriter<F> newManifestWriter(PartitionSpec spec);

protected abstract ManifestReader<F> newManifestReader(ManifestFile manifest);
```

One caveat is calling this method to initialize an instance field. It is considered a bad practice but implementations will be stateless, so it will work. We could pass `Supplier<Set<F>>` but not sure it is better. In either case, we need to find a way not to have both sets of files here. It will also reduce the number of changes.

Reply:
I'm also not a big fan of having `FilesToDeleteHolder` and it was just an experiment to see how the changes would look, but it is indeed probably better to have an abstract `Set<F> newFileSet()` method. I've refactored the code and it should look cleaner now, thanks for the suggestion